csrc/quantization/cutlass_w8a8/scaled_mm_entry.cu

```diff
@@ -168,6 +182,47 @@ void cutlass_scaled_mm(torch::Tensor& c, torch::Tensor const& a,
       version_num);
 }
 
+void cutlass_grouped_mm(
+    torch::Tensor& out_tensors, torch::Tensor const& a_tensors,
+    torch::Tensor const& b_tensors, torch::Tensor const& a_scales,
+    torch::Tensor const& b_scales, torch::Tensor const& expert_offsets,
+    torch::Tensor const& problem_sizes, torch::Tensor const& a_strides,
+    torch::Tensor const& b_strides, torch::Tensor const& c_strides) {
+  int32_t version_num = get_sm_version_num();
+#if defined ENABLE_SCALED_MM_SM90 && ENABLE_SCALED_MM_SM90
+  cutlass_grouped_mm_sm90(out_tensors, a_tensors, b_tensors, a_scales, b_scales,
+                          expert_offsets, problem_sizes, a_strides, b_strides,
+                          c_strides);
+  return;
+#endif
+  TORCH_CHECK_NOT_IMPLEMENTED(
+      false,
+      "No compiled cutlass_grouped_mm for a compute capability less than "
+      "CUDA device capability: ",
+      version_num);
+}
+
+void get_grouped_mm_data(const torch::Tensor& topk_ids,
+                         torch::Tensor& expert_offsets,
+                         torch::Tensor& problem_sizes1,
+                         torch::Tensor& problem_sizes2, torch::Tensor& arg_sort,
+                         torch::Tensor& arg_sort_prim,
+                         const int64_t num_experts, const int64_t n,
+                         const int64_t k) {
+  int32_t version_num = get_sm_version_num();
+#if defined ENABLE_SCALED_MM_SM90 && ENABLE_SCALED_MM_SM90
```

Comment:
`get_grouped_mm_data` seems to be generic cuda kernels, any reason this has to be compiled only for SM90? These are helpful for all cuda versions AFAI can tell.

Reply:
I removed the requirement and moved the function to `csrc/quantization/cutlass_w8a8/moe_data.cu`.